Dockerfile

```diff
@@ -1,11 +1,25 @@
-FROM rust:1.34.0-slim
+FROM debian:buster-slim
 
 RUN apt-get update
 RUN apt-get install -y clang cmake
 RUN apt-get install -y libsnappy-dev
+RUN apt-get install -y curl
 
 RUN adduser --disabled-login --system --shell /bin/false --uid 1000 user
 
+ARG RUST_VERSION=1.34.0
+ENV RUSTUP_HOME /usr/local/rustup
+ENV CARGO_HOME /usr/local/cargo
+ENV PATH $CARGO_HOME/bin:$PATH
+
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- \
+  -y \
+  --verbose \
+  --profile minimal \
+  --default-toolchain $RUST_VERSION
+
+RUN chmod -R a+w $RUSTUP_HOME $CARGO_HOME
```

Comment:
This looks strange, is it a standard thing to do in docker?

Reply:
No, it's not a particularly standard thing to do in Docker, normally you would compile a binary then copy it to `usr/local/bin` or similar.

Not very familiar with Cargo but it seems to want it's own build dir with other data that you then add to your path.

Without changing the permissions the `user` account doesn't have permission to write to `$CARGO_HOME` during installation.

The official Rust images do it like this too: https://github.com/rust-lang/docker-rust/blob/8bab191937fcf23569d3a3c31103c1c6f7f2947e/1.42.0/buster/slim/Dockerfile#L30

I guess we could just build as root in `/root/.cargo/bin` and then copy the `electrs` binary to `/usr/local/bin` which should allow `user` to execute it.